### PR TITLE
[build] Switch to self-hosted GTest/GMock

### PR DESCRIPTION
**Pull request by vshampor**

### CODE REVIEW

File: samples/metrics_monitor/CMakeLists.txt [hunk 1]
```diff
@@ -22,36 +22,24 @@ list( APPEND LIBS cttmetrics)
...
-    list( APPEND sources ${srcs})
-    list( APPEND LIBS
-      pthread
-      rt
-      gtest )
-
-    set(DEPENDENCIES libdrm)
-
-    make_executable( test_monitor none )
+    install(TARGETS test_monitor RUNTIME DESTINATION ${MFX_APPS_DIR}/tests)
```
Comment (dvrogozh) [inline samples/metrics_monitor/CMakeLists.txt]: This line is a partial fix for #752. Can you, please, extract this in a separate commit and separate PR? And please, add install commands for the other targets from #752.

File: CMakeLists.txt [hunk 2]
```diff
@@ -90,7 +90,6 @@ include( ${BUILDER_ROOT}/FindMFX.cmake )
 include( ${BUILDER_ROOT}/FindPackages.cmake )
```
Comment (dvrogozh) [inline CMakeLists.txt]: Add:
`Fixes: #836`
to the commit message

File: CMakeLists.txt.in [hunk 3]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 2.8.2)
```
Comment (dvrogozh) [inline CMakeLists.txt.in]: I think you mentioned that gtest fails to build w/ cmake-2.8? What is the real requirement? Do we have any hint in gtest readme?

File: CMakeLists.txt.in [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+project(googletest-download NONE)
+
+include(ExternalProject)
+ExternalProject_Add(googletest
+  GIT_REPOSITORY    https://github.com/google/googletest.git
+  GIT_TAG           master
```
Comment (dvrogozh) [inline CMakeLists.txt.in]: I think we should use some tagged release.

File: CMakeLists.txt.in (hunk 3 above)
Comment (vshampor) [reply CMakeLists.txt.in]: Looks like the requirement is for the CMake version to be at least 3.0:
https://github.com/google/googletest/issues/1786

Comment (uartie) [reply CMakeLists.txt.in]: Have you tried with the `release-1.8.0` tag?

Comment (vshampor) [reply CMakeLists.txt.in]: Indeed, the release-1.8.0 tag version allows to configure the build with CMake 2.8.12 as well.

File: CMakeLists.txt.in (hunk 4 above)
Comment (vshampor) [reply CMakeLists.txt.in]: Done

File: CMakeLists.txt.in (hunk 3 above)
Comment (vshampor) [reply CMakeLists.txt.in]: However, the release-1.8.0 does not compile - seems to be an internal bug in GTest. We will have to switch to CMake3 after all.

Comment (uartie) [reply CMakeLists.txt.in]: What is the compile error?  For other projects, like intel-vaapi-driver, libva-utils and libyami release-1.8.0 compiles fine in multiple environments... of course, those projects use automake, though.

Comment (dmitry-gurulev) [reply CMakeLists.txt.in]: @vshampor,

JFI: release-1.8.x is pre-C++11 and uses a technic of functions overload bunch (_ElementsAre_ matcher is eg.) instead of variadic - this leads to slower compilation.

Comment (vshampor) [reply CMakeLists.txt.in]: See http://mediasdk.intel.com/buildbot/#/builders/6/builds/1936 for errors. This is reproduced locally on my machine running CentOS 7.4/CMake 2.8.12 as well.

**Comment by vshampor** [comment]
To be reopened via a fork PR

File: CMakeLists.txt.in (hunk 3 above)
Comment (uartie) [reply CMakeLists.txt.in]: I see, it fails in gmock.  I've never tried compiling gmock, sorry :/

Comment (uartie) [reply CMakeLists.txt.in]: @vshampor looks like the `error: ‘mock.MockCallObj::gmock_dlclose’ does not have class type` errors are due to incorrect gmock usage.  https://github.com/google/googletest/blob/release-1.8.0/googlemock/test/gmock-spec-builders_test.cc#L225 has examples of how the MOCK_METHODn is used with EXPECT_CALL.

Perhaps this will help improve your forked PR.

Comment (dvrogozh) [reply CMakeLists.txt.in]: @uartie, @dmitry-gurulev  : we are working to clean up branches which we have in this repository and started to prohibit PR from within major repo. This PR is one of these that's why @vshampor closed it and opened new one: #856. I apologize for the inconveniences, but could you, please, continue discussion in #856?
